lib/src/excel.dart

```diff
@@ -1,36 +1,128 @@
 @JS()
 library excel;
 
+import 'package:fixnum/fixnum.dart';
 import 'package:js/js.dart';
 
 import 'office_js.dart';
+import 'proto/insertion_order.pb.dart';
 
 /// Service class that provides dart functions to interact with Excel.
 class ExcelDart {
-  ExcelDart._private();
+  static final _startAddressInt = 'A'.codeUnitAt(0);
+  static final _fontName = 'Roboto';
+  static final _fontSize = 12;
+  static final _horizontalAlignment = 'Center';
+  static final _borderStyle = 'Continuous';
 
-  static final ExcelDart _singleton = ExcelDart._private();
+  static final _tableHeader = [
+    'Insertion Order ID',
+    'Advertiser ID',
+    'Campaign ID',
+    'Display Name',
+    'Entity Status',
+    'Update Time',
+    'Pacing Period',
+    'Pacing Type',
+    'Daily Max Micros',
+    'Daily Max Impressions',
+    'Budget Unit',
+    'Automation Type',
+    'Budget',
+    'Start Date',
+    'End Date'
+  ];
 
-  factory ExcelDart() {
-    return _singleton;
-  }
-
-  /// Waits for Office APIs to be ready and then executes function [_populate].
-  void exec() async {
+  /// Waits for Office APIs to be ready and then populates the spreadsheet with
+  /// entries in the [insertionOrderList]
+  static void populate(List insertionOrderList) async {
     await Office.onReady(allowInterop((info) async {
-      await ExcelJS.run(allowInterop(_populate));
+      await ExcelJS.run(allowInterop((context) {
+        // Grabs the current worksheet
+        final sheet = context.workbook.worksheets.getActiveWorksheet();
```

Comment:
consider: making a new sheet to avoid problems of overwriting/conflicting with existing data